src/coreclr/src/debug/daccess/request.cpp

```diff
@@ -4741,3 +4798,202 @@ HRESULT ClrDataAccess::GetBreakingChangeVersion(int* pVersion)
     *pVersion = SOS_BREAKING_CHANGE_VERSION;
     return S_OK;
 }
+
+HRESULT ClrDataAccess::GetObjectComWrappersData(CLRDATA_ADDRESS objAddr, CLRDATA_ADDRESS *rcw, unsigned int count, CLRDATA_ADDRESS *mowList, unsigned int *pNeeded)
+{
+#ifdef FEATURE_COMWRAPPERS
+    if (objAddr == 0 )
+    {
+        return E_INVALIDARG;
+    }
+
+    if (count > 0 && mowList == NULL)
+    {
+        return E_INVALIDARG;
+    }
+
+    SOSDacEnter();
+    auto ComWrapperCallback = [](void *mocw, void *additionalData)
+    {
+        CQuickArrayList<CLRDATA_ADDRESS> *comWrappers = (CQuickArrayList<CLRDATA_ADDRESS>*)additionalData;
+
+        comWrappers->Push(TO_CDADDR(mocw));
+
+    };
+
+    // We could bail due to missing memory, want to return S_FALSE in that case
+    hr = S_FALSE;
+    if (pNeeded != NULL)
+    {
+        *pNeeded = 0;
+    }
+
+    if (rcw != NULL)
+    {
+        *rcw = 0;
+    }
+
+    PTR_SyncBlock pSyncBlk = PTR_Object(TO_TADDR(objAddr))->PassiveGetSyncBlock();
+    if (pSyncBlk != NULL)
+    {
+        PTR_InteropSyncBlockInfo pInfo = pSyncBlk->GetInteropInfoNoCreate();
+        if (pInfo != NULL)
+        {
+            if (rcw != NULL)
+            {
+                *rcw = PTR_CDADDR(pInfo->m_externalComObjectContext);
+            }
+
+            CQuickArrayList<CLRDATA_ADDRESS> comWrappers;
+            pInfo->IterateComWrappers(ComWrapperCallback, (void *)&comWrappers);
+
+            if (pNeeded != NULL)
+            {
+                *pNeeded = (unsigned int)comWrappers.Size();
+            }
+
+            for (SIZE_T pos = 0; pos < comWrappers.Size(); ++pos)
+            {
+                if (pos >= count)
+                {
+                    // Already set S_FALSE above
+                    break;
+                }
+
+                mowList[pos] = comWrappers[pos];
+            }
+
+            // hr is set to S_FALSE above because of the possiblity that the memory is missing and we 
+            // could hit the catch, the only codepath we want to return S_OK is if we successfully
+            // return data. Even if the memory exists but the array is too small we return S_FALSE too
+            hr = S_OK;
+        }
+    }
+
+    SOSDacLeave();
+    return hr;
+#else // FEATURE_COMWRAPPERS
+    return E_NOTIMPL;
+#endif // FEATURE_COMWRAPPERS
+}
+    
+HRESULT ClrDataAccess::IsComWrappersCCW(CLRDATA_ADDRESS ccw, BOOL *isComWrappersCCW)
+{
+#ifdef FEATURE_COMWRAPPERS
+    if (ccw == 0)
+    {
+        return E_INVALIDARG;
+    }
+
+    SOSDacEnter();
+    
+    if (isComWrappersCCW != NULL)
```

Comment:
Maybe return S_FALSE if the `ccw` is not a com wrapper list thing?  (Just a thought, not pushback on this design.)

Reply:
Sure, I think that makes sense